components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { ArrowLeft, ArrowRight } from 'react-feather';
+
+import { PaginationProps } from '@/components/Pagination/Pagination.types';
+
+const Pagination = ({
+  current,
+  total,
+  onPageClick,
+  limit = 1
+}: PaginationProps) => {
+  const [isPrevAvailable, setIsPrevAvailable] = useState(false);
+  const [isNextAvailable, setIsNextAvailable] = useState(false);
+  const [pages, setPages] = useState<number[]>([current]);
+  const [centerPage, setCenterPage] = useState(0);
+
+  const calculatePages = (
+    page: number,
+    total: number,
+    limit: number
+  ): number[] => {
+    const pages: number[] = [];
+    const maxPage = Math.ceil(total / limit);
+    const minPage = 1;
+    const maxPages = 5;
+    const minPageNumber = Math.max(minPage, page - Math.floor(maxPages / 2));
+    const maxPageNumber = Math.min(maxPage, page + Math.floor(maxPages / 2));
+    const pagesCount = maxPageNumber - minPageNumber + 1;
+    const pagesToAdd = maxPages - pagesCount;
+    const minPageNumberToAdd = Math.max(minPage, minPageNumber - pagesToAdd);
+    const maxPageNumberToAdd = Math.min(maxPage, maxPageNumber + pagesToAdd);
+    for (let i = minPageNumberToAdd; i <= maxPageNumberToAdd; i++) {
+      pages.push(i);
+    }
+    setCenterPage(
+      pages.length % 2 === 0 ? pages.length / 2 : (pages.length + 1) / 2
+    );
+    if (pages.length === 0 && page > 0) {
+      pages.push(page);
+    }
+    return pages;
+  };
+
+  useEffect(() => {
+    setIsPrevAvailable(current > 1);
+    setIsNextAvailable(current * limit < total);
+    setPages(calculatePages(current, total, limit));
+  }, [current, limit, total]);
+
+  return (
+    <div className="flex w-full items-center justify-between border-t py-4 px-2">
+      <div className="flex flex-1 justify-start">
```

Comment:
it seems like 

```
<beforeButton>
{children}
<nextButton> 

```
to me. 

and also,
maybe we can add showFirstButton showLastButton props too, let's say we have 1000 pages... it would be easier to go 1, or last.

Reply:
I'm adding the 1st one to the list.

For the 2nd, I think the current infrastructure provides it. I'll test and report as the case may be, or add to the list.